sormas-app/app/src/main/java/de/symeda/sormas/app/pathogentest/edit/PathogenTestEditActivity.java

```diff
@@ -92,39 +93,43 @@ public void saveData() {
-			DiseaseVariant newDiseaseVariant = pathogenTestToSave.getTestedDiseaseVariant();
-			if (pathogenTestToSave.getTestResult() == PathogenTestResultType.POSITIVE
-				&& pathogenTestToSave.getTestResultVerified()
-				&& !DataHelper.equal(newDiseaseVariant, caseDiseaseVariant)) {
-
-				String heading = I18nProperties.getString(Strings.headingUpdateCaseWithNewDiseaseVariant);
-				String subHeading = I18nProperties.getString(Strings.messageUpdateCaseWithNewDiseaseVariant);
-				int positiveButtonTextResId = R.string.yes;
-				int negativeButtonTextResId = R.string.no;
-
-				ConfirmationDialog dlg = new ConfirmationDialog(this, heading, subHeading, positiveButtonTextResId, negativeButtonTextResId);
-				dlg.setCancelable(false);
-				dlg.setNegativeCallback(() -> {
-					save(pathogenTestToSave, associatedCase);
-				});
-				dlg.setPositiveCallback(() -> {
-					associatedCase.setDiseaseVariant(newDiseaseVariant);
-					try {
-						DatabaseHelper.getCaseDao().updateOrCreate(associatedCase);
-					} catch (SQLException | java.sql.SQLException e) {
-						Log.e(getClass().getSimpleName(), "Could not update case: " + associatedCase.getUuid());
-						throw new RuntimeException(e);
-					}
+		if (pathogenTestToSave.getSample() != null) {
+			final Case associatedCase = pathogenTestToSave.getSample().getAssociatedCase();
+
+			if (associatedCase != null) {
+				DiseaseVariant caseDiseaseVariant = associatedCase.getDiseaseVariant();
+				DiseaseVariant newDiseaseVariant = pathogenTestToSave.getTestedDiseaseVariant();
+				if (pathogenTestToSave.getTestResult() == PathogenTestResultType.POSITIVE
+					&& pathogenTestToSave.getTestResultVerified()
+					&& !DataHelper.equal(newDiseaseVariant, caseDiseaseVariant)) {
+
+					String heading = I18nProperties.getString(Strings.headingUpdateCaseWithNewDiseaseVariant);
+					String subHeading = I18nProperties.getString(Strings.messageUpdateCaseWithNewDiseaseVariant);
+					int positiveButtonTextResId = R.string.yes;
+					int negativeButtonTextResId = R.string.no;
+
+					ConfirmationDialog dlg = new ConfirmationDialog(this, heading, subHeading, positiveButtonTextResId, negativeButtonTextResId);
+					dlg.setCancelable(false);
+					dlg.setNegativeCallback(() -> {
+						save(pathogenTestToSave, associatedCase);
+					});
+					dlg.setPositiveCallback(() -> {
+						associatedCase.setDiseaseVariant(newDiseaseVariant);
+						try {
+							DatabaseHelper.getCaseDao().updateOrCreate(associatedCase);
+						} catch (SQLException | java.sql.SQLException e) {
+							Log.e(getClass().getSimpleName(), "Could not update case: " + associatedCase.getUuid());
+							throw new RuntimeException(e);
+						}
+						save(pathogenTestToSave, associatedCase);
+					});
+					dlg.show();
+				} else {
 					save(pathogenTestToSave, associatedCase);
-				});
-				dlg.show();
-			} else {
-				save(pathogenTestToSave, associatedCase);
+				}
 			}
+		} else if (pathogenTestToSave.getEnvironmentSample() != null) {
```

Comment:
there is no need for this condition, just save

Reply:
removed